src/core/server/elasticsearch/scoped_cluster_client.test.ts

```diff
@@ -134,6 +146,33 @@ describe('#callAsCurrentUser', () => {
     expect(internalAPICaller).not.toHaveBeenCalled();
   });
 
+  test('callAsCurrentUser allows passing additional headers', async () => {
+    const mockResponse = { data: 'response' };
+    scopedAPICaller.mockResolvedValue(mockResponse);
+    await expect(
+      clusterClient.callAsCurrentUser('security.authenticate', {
+        some: 'some',
+        headers: { additionalHeader: 'Oh Yes!' },
+      })
+    ).resolves.toBe(mockResponse);
+    expect(scopedAPICaller).toHaveBeenCalledTimes(1);
+    expect(scopedAPICaller).toHaveBeenCalledWith(
+      'security.authenticate',
+      { some: 'some', headers: { one: '1', additionalHeader: 'Oh Yes!' } },
+      undefined
+    );
+    scopedAPICaller.mockClear();
+  });
+
+  test('callAsCurrentUser cannot override default headers', async () => {
+    const mockErrorResponse = new Error('Cannot override default headers [one].');
```

Comment:
naming is confusing, you don't mock anything. 
```js
    const withHeaderOverride = async () =>
      clusterClient.callAsCurrentUser('security.authenticate', { headers: { one: 'OVERRIDE' } });
    await expect(withHeaderOverride()).rejects.toThrowErrorMatchingInlineSnapshot(
      `"Cannot override default header one."`
    );
```

Reply:
and test doesn't match https://github.com/elastic/kibana/pull/34610/files#diff-e671f92de09a3e2c73ef0bc0b1ff35cbR84